### ISSUE TITLE
behavior should be array?

### ISSUE THREAD

**Issue by zandbelt** [openid-certification/oidctest#31]
I ran into an issue with the "behavior" element in the JSON test descriptions: they are strings but the operations on them assume arrays, as a lot of them look like
```
        if "iat" in self.behavior_type:  # missing iat claim
```
etc.

The `self.behavior_type` initialization suggests it is stored as an array (https://github.com/openid-certification/oidctest/blob/master/src/oidctest/rp/provider.py#L170) or even object (https://github.com/openid-certification/oidctest/blob/master/src/oidctest/rp/provider.py#L104) but at assignment time it makes it the type of whatever the test description value is set to (https://github.com/openid-certification/oidctest/blob/master/src/oidctest/cp/op_handler.py#L136)
```
            op.behavior_type = _tc["behavior"]
            op.server.behavior_type = _tc["behavior"]
```
Since the current tests define behavior as a string e.g.
```
  "behavior": "aud",
```
the behavior matching code will actually test for a *substring* in a string instead of a string in an array. That's dangerous as I named my behavior:
```
  "behavior": "initiate_login_uri",
```
which unexpectedly matches `iat` as well.
***
In summary, I believe all test descriptions should change to use an array as in:
```
  "behavior": [ "initiate_login_uri" ],
```
or else all code handling it would need to change to avoid future name clashes like the one I had.

**Comment by rohe** [comment]
Actually, initially behavior_type was a dictionary.
To allow for finer grained behaviorial description.

**Comment by rohe** [comment]
Could probably be content with a list at this time. 
Have not got a use case when a dict is needed.

**Comment by rohe** [comment]
Solved by
https://github.com/rohe/oidctest/commit/838903da8077efebeb1be0bdfd34fdc2e7fe9cd6 and
https://github.com/rohe/oidctest/commit/1d31c69d4461e79a49b753a1dcd73caa492dc3ba

**Comment by zandbelt** [comment]
that's fixed in the current codebase
